src/illuminatio/illuminatio.py

```diff
@@ -154,24 +173,18 @@ def run(
     if outfile:
         # write output
         LOGGER.info("Writing results to file %s", outfile)
-        _, extension = path.splitext(outfile)
         file_contents = {
             "cases": results,
             "runtimes": runtimes,
             "results": additional_data,
         }
         file_contents["runtimes"]["runners"] = test_runtimes
-        file_contents["runtimes"]["generator"] = gen_run_times
-        if extension in [".yaml", ".yml"]:
-            with open(outfile, "w") as out:
-                yaml.dump(file_contents, out, default_flow_style=False)
-        elif extension == ".json":
-            with open(outfile, "w") as out:
-                json.dump(file_contents, out)
-        else:
-            logging.error(
-                "Output format %s not supported! Aborting write to file.", extension
-            )
+        if gen_run_times:
+            file_contents["runtimes"]["generator"] = gen_run_times
```

Comment:
After sleeping over it I would prefer to set `gen_run_times` to `0` if the test cases are provided and always set the field in the output file, this will make the parsing of the output file less fragile